cli/printers/table_printer.py

```diff
@@ -0,0 +1,123 @@
+from cli.printers.base_printer import BasePrinter
+
+
+class TablePrinter(BasePrinter):
+    RED_COLOR_NAME = 'red'
+    WHITE_COLOR_NAME = 'white'
+    GREEN_COLOR_NAME = 'green'
+
+    scan_id: str
+
+    def __init__(self, context: click.Context):
+        super().__init__(context)
+        self.scan_id = context.obj.get('scan_id')
+
+    def print_results(self, results: List[DocumentDetections]):
+        click.secho(f"Scan Results: (scan_id: {self.scan_id})")
+
+        if not results:
+            click.secho("Good job! No issues were found!!! 👏👏👏", fg=self.GREEN_COLOR_NAME)
+            return
+
+        detections_per_detection_type_id = self._extract_detections_per_detection_type_id(results)
+
+        self._print_detection_per_detection_type_id(detections_per_detection_type_id)
+
+        if self.context.obj.get('report_url'):
+            click.secho(f"Report URL: {self.context.obj.get('report_url')}")
+
+    def _extract_detections_per_detection_type_id(self, results: List[DocumentDetections]):
+        detections_per_detection_type_id = {}
+
+        for document_detection in results:
+            for detection in document_detection.detections:
+                if detection.detection_type_id not in detections_per_detection_type_id:
+                    detections_per_detection_type_id[detection.detection_type_id] = []
+                detections_per_detection_type_id[detection.detection_type_id].append(detection)
+
+        return detections_per_detection_type_id
+
+    def _print_detection_per_detection_type_id(self, detections_per_detection_type_id: Dict[str, Detection]):
+        for detection_type_id in list(detections_per_detection_type_id):
+
+            detections = detections_per_detection_type_id[detection_type_id]
+            if len(detections) == 0:
+                continue
+
+            headers = [REPOSITORY_COLUMN] if self._is_repository() else []
+
+            if detection_type_id == PACKAGE_VULNERABILITY_POLICY_ID:
+                headers = [SEVERITY_COLUMN] + headers
+                headers = headers + PREVIEW_DETECTIONS_COMMON_HEADERS[:]
+                headers.append(UPGRADE_COLUMN)
+                self._print_table_detections(detections,
+                                             headers,
+                                             self._get_upgrade_package_vulnerability,
+                                             "Dependencies Vulnerabilities")
+
+            if detection_type_id == LICENSE_COMPLIANCE_POLICY_ID:
+                headers = headers + PREVIEW_DETECTIONS_COMMON_HEADERS[:]
+                headers.append(LICENSE_COLUMN)
+                self._print_table_detections(detections,
+                                             headers,
+                                             self._get_license,
+                                             "License Compliance")
+
+    def _print_table_detections(self, detections: List[Detection], headers: List[str],
```

Comment:
instead of pass here the get_row func just generate the rows before this func and pass here the final rows

Reply:
Fixed